packages/workflow/src/records/fhir.ts

```diff
@@ -807,7 +885,130 @@ export async function sendBundleToHearth(
     )
   }
 
-  return res.json()
+  const responseBundle: TransactionResponse = await res.json()
+
+  const ok = responseBundle.entry.every((e) =>
+    e.response.status.startsWith('2')
+  )
+  if (!ok) {
+    throw new Error(
+      'Hearth was unable to create/update all the entires in the bundle'
+    )
+  }
+
+  return responseBundle
+}
+
+function findSavedReference(
+  temporaryReference: URNReference,
+  resourceBundle: Bundle,
+  responseBundle: TransactionResponse
+): URLReference | null {
+  const indexInResponseBundle = resourceBundle.entry.findIndex(
+    (entry) => entry.fullUrl === temporaryReference
+  )
+  if (indexInResponseBundle === -1) {
+    return null
+  }
+  return responseBundle.entry[indexInResponseBundle].response.location
+}
+
+function toSavedComposition(
```

Comment:
Would the functions in this file make sense in the `commons` package?

Reply:
I was thinking the same but they should not be needed anywhere other than workflow which defeats the purpose